### PR TITLE
fix: dump gateway errors during TLS handshake

### PR DESCRIPTION
**Pull request by troian**
Signed-off-by: Artur Troian <troian.ap@gmail.com>

---
name: Pull Request
about: Submit a pull request
title: ''
labels: ''
assignees: '@boz @gosuri'

---

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/ovrclk/akash/blob/master/CONTRIBUTING.md))
- [ ] My issue has received approval from the maintainers or lead with the `design/approved` label

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I've read the [CONTRIBUTION](https://github.com/ovrclk/akash/blob/master/CONTRIBUTING.md) guide
- [ ] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: provider/gateway/utils/utils.go [hunk 1]
```diff
@@ -27,24 +28,34 @@ func NewServerTLSConfig(ctx context.Context, certs []tls.Certificate, cquery cty
 				}
 
 				cert, err := x509.ParseCertificate(certificates[0])
+				defer func() {
+					if err != nil {
+						ctx.Value("log").(log.Logger).Error("VerifyPeerCertificate failed", "err", err.Error())
```
[the remark this answers is not in the document]
Comment (boz) [reply provider/gateway/utils/utils.go:33]: also, I *believe* that `err` will only be what you think it is here if it is declared like this:

```go
func(certificates [][]byte, _ [][]*x509.Certificate) (err error) {
```

**Review by boz — CHANGES_REQUESTED** [review]
some comments inline - make it a bit less brittle and make it something that can be used elsewhere (`ctxlog`)

**Comment by hydrogen18** [comment]
@troian can we close this?

**Comment by troian** [comment]
yah

**Pull request closed without merging**